packages/ui/src/forum/components/PostList/PostEditor.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useMemo, useState } from 'react'
+import styled from 'styled-components'
+
+import { ButtonGhost, ButtonPrimary, ButtonsRow } from '@/common/components/buttons'
+import { CKEditor } from '@/common/components/CKEditor'
+import { useModal } from '@/common/hooks/useModal'
+import { EditPostModalCall } from '@/forum/modals/PostActionModal/EditPostModal'
+import { ForumPost } from '@/forum/types'
+
+interface Props {
+  post: ForumPost
+  onCancel: () => void
+}
+
+export const PostEditor = ({ post, onCancel }: Props) => {
+  const [newText, setNewText] = useState(post.text)
+  const { showModal } = useModal()
+  const isTextChanged = useMemo(() => post.text !== newText, [newText])
+
+  return (
+    <EditorWrap>
+      <EditorMemo setNewText={setNewText} initialText={post.text} />
+      <ButtonsRow>
+        <ButtonGhost size="medium" onClick={onCancel}>
+          Cancel
+        </ButtonGhost>
+        <ButtonPrimary
+          size="medium"
+          onClick={() => showModal<EditPostModalCall>({ modal: 'EditPost', data: { newText, post } })}
+          disabled={!isTextChanged}
+        >
+          Save
+        </ButtonPrimary>
+      </ButtonsRow>
+    </EditorWrap>
+  )
+}
+
+interface MemoEditorProps {
+  setNewText: (t: string) => void
+  initialText: string
+}
+
+const EditorMemo = React.memo(({ setNewText, initialText }: MemoEditorProps) => (
+  <CKEditor
+    inline
+    id="editor"
+    onChange={(_, editor) => setNewText(editor.getData())}
+    onReady={(editor) => editor.setData(initialText)}
+  />
+))
```

Comment:
Wrapped the CKEditor in a memo here, because otherwise rerendering the `useState` hook for `newText` caused the whole component to rerender, upon which a new instance of the editor was created, resulting in an [error](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/support/error-codes.html#error-editor-source-element-already-used).
This is probably a hack, but it was the only way I've found to work around an issue. Maciej has a much better knowledge of CKEditor so I'll ask him if we could do it better.